WebdevPeriod3/wwwroot/js/subsystemlist.js

```diff
@@ -0,0 +1,115 @@
+﻿'use strict';
+
+function getButtonRow(buttonEl) { 
+  buttonEl.closest('input'); 
+}
+
+function debounce(callback, wait, immediate = false) {
+  let timeout = null 
+  
+  return function() {
+    const callNow = immediate && !timeout
+    const next = () => callback.apply(this, arguments)
+    
+    clearTimeout(timeout)
+    timeout = setTimeout(next, wait)
+
+    if (callNow) {
+      next()
+    }
+  }
+}
+
+let ctrl = null;
+
+async function search(term) {
+  if (ctrl) ctrl.abort();
+  ctrl = new AbortController();
+
+  const products = await fetch(`../products?searchterm=${term}`, {
+    signal: ctrl.signal
+  })
+    .then((res) => res.json())
+    .catch((ex) => (ex.name === "AbortError" ? null : Promise.reject(ex)))
+    .finally(() => {
+      ctrl = null;
+    });
+
+    console.log(products);
+
+  return products?.filter((product) =>
+    product.name.toLowerCase()
+  );
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+
+  let terms;
+
+    document.getElementById("search-input").addEventListener("input", debounce(async (e) => {
+    if (e.target.value)
```

Comment:
Naked if statements are bad practice, if someone wants to adjust your code later on they have to add brackets everywhere. Especially here you should use brackets, you might get away with the other occurences.